src/errors.rs

```diff
@@ -51,5 +54,34 @@ quick_error! {
     }
 }
 
+#[cfg(not(feature = "protobuf"))]
+quick_error! {
+    /// The error types for prometheus.
+    #[derive(Debug)]
+    pub enum Error {
+        /// A duplicate metric collector has already been registered.
+        AlreadyReg {
+            description("duplicate metrics collector registration attempted")
+        }
+        /// The label cardinality was inconsistent.
+        InconsistentCardinality(expect: usize, got: usize) {
+            description("inconsistent label cardinality")
+            display("expect {} label values, but got {}", expect, got)
+        }
+        /// An error message which is only a string.
+        Msg(msg: String) {
+            description(msg)
+            display("Error: {}", msg)
+        }
+        /// An error containing a [`std::io::Error`].
+        Io(err: IoError) {
+            from()
+            cause(err)
+            description(err.description())
+            display("Io {}", err)
+        }
+    }
+}
```

Comment:
This duplication of the entire error declaration is quite unfortunate. I can't think how to avoid it offhand though :-/